hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/metrics/OzoneMutableStat.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.metrics;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.hadoop.metrics2.MetricsInfo;
+import org.apache.hadoop.metrics2.MetricsRecordBuilder;
+import org.apache.hadoop.metrics2.lib.MutableStat;
+import org.apache.hadoop.metrics2.util.SampleStat;
+import org.apache.hadoop.util.Time;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import static org.apache.hadoop.metrics2.lib.Interns.info;
+
+/**
+ * A convenient mutable metric for throughput measurement.
+ * (non-synchronized version of org.apache.hadoop.metrics2.lib.MutableStat)
+ */
+public class OzoneMutableStat extends MutableStat {
```

Comment:
Is it supposed to extend `MutableStat`?  This duplicates all member variables.  Can we copy all necessary classes without subclassing?

Reply:
All metrics should extend MutableMetric. I will change extending MutableStat to MutableMetric.